gdb-server/src/bin.rs

```diff
@@ -30,6 +30,15 @@ struct Opt {
         help = "Use this flag to override the default GDB connection string (localhost:1337)."
     )]
     gdb_connection_string: Option<String>,
+    #[structopt(name = "list", long = "list", help = "list available debug probes")]
+    list: bool,
+    #[structopt(
+        name = "debug probe index",
+        long = "probe-index",
+        short = "n",
+        help = "select index of debug probe to use"
+    )]
+    probe_index: Option<usize>,
```

Comment:
On other tools we do VID/PID based matching which is less cumbersome to use. I think we should use this here too. I am ok with the probe index but if we already make a change, we might as well do it right :)

Reply:
Well, there is also the possibility of having multiple probes with identical vid/pid/serial. Should I keep the index, and add a separate vid/pid option?

Reply:
Hmm agreed, that can happen if they disregard the spec :(

I think it would be great if you added that :)